src/data_scraper/time_helpers.py

```diff
@@ -68,4 +70,44 @@ def str_to_timestamp(string, format: ['date', 'exact_time']):
     elif format == 'exact_time':
         return int(round(dt.datetime.strptime(string, "%Y-%m-%d %H:%M:%S").timestamp())) * 1000
     else:
-        raise Exception('Format has to be either "date" or "exact_time".')
\ No newline at end of file
+        raise Exception('Format has to be either "date" or "exact_time".')
+
+
+def slice_timestamps_in_chunks(start_timestamp, end_timestamp):
+    """
+    Slice the full timeframe between start and end timestamps into weekly chunks.
+    The purpose is to avoid API crashes when loading large timeframes of data.
+    If the total timeframe does not sum up to a round number of weeks, the last week is kept as it is.
+    All starting timestamps in chunks are shifted by the production lookback window.
+    :param start_timestamp: timestamp in ms
+    :param end_timestamp: timestamp in ms
+    :return: [[chunk_start, chunk_end], [chunk_start, chunk_end], ...]
+    """
+    one_week_ms = 1000 * 60 * 60 * 24 * 7  # ms * s * m * h * d
+    full_weeks = (end_timestamp - start_timestamp) // one_week_ms
+    total_weeks = (end_timestamp - start_timestamp) / one_week_ms
+
+    # If total time frame is less than one week, return the original timestamps with a lookback window
+    if total_weeks < 1:
+        return [[get_production_start_timestamp(start_timestamp), end_timestamp]]
```

Comment:
WDYT about passing directly the actual start_time after doing `get_production_start_timestamp(start_timestamp)`?  I think it makes more sense to put this function in `load_from_disk` or even in the training service. This time helpers class shouldn't care about this I think.

Reply:
totally makes sense, it's like you suggest in the latest commit